src/Cake.Common/Build/TeamCity/Data/TeamCityPullRequestInfo.cs

```diff
@@ -11,6 +11,39 @@ namespace Cake.Common.Build.TeamCity.Data
     /// </summary>
     public class TeamCityPullRequestInfo : TeamCityInfo
     {
+        private static bool InferIsPullRequest(string gitReferenceName)
+        {
+            var branchSlices = gitReferenceName.Split('/');
```

Comment:
Should check & return if `gitReferenceName` is null or empty.

Reply:
done